sdk/cosmosdb/cosmos/src/client/Database/Database.ts

```diff
@@ -127,6 +139,12 @@ export class Database {
     const offer = response.result[0]
       ? new Offer(this.client, response.result[0].id, this.clientContext)
       : undefined;
-    return new OfferResponse(response.result[0], response.headers, response.code, offer);
+    return new OfferResponse(
```

Comment:
Handle empty response

Reply:
So here `response` is an internally created object. Which always returns a fully formed object. [ref](https://github.com/Azure/azure-sdk-for-js/blob/7cd790b1b5437841b289beef74d9814853d914bc/sdk/cosmosdb/cosmos/src/request/RequestHandler.ts#L147)